### PR TITLE
Add tests and support for debian 10

### LINKED ISSUE #348: haproxy default package version is 1.5
### What's the issue
Found this when converging a test recipe so I had a kitchen to play around with. Found out that the default haproxy package install version is 1.5 (confirmed on centos 7, haven't replicated on other platforms yet). 

### Why does it matter
A lot of tests use `haproxy_install 'package'` and 1.5 is `critical patches only`

### What should we do
Explicitly set the haproxy package version

### PR DESCRIPTION
**Pull request by JohnRoesler**
### Description

- Adds circleci tests against debian 10 and notes support in readme
- Removes `package-centos-7` test

### Issues Resolved
closes #348 haproxy default package version is 1.5 on centos-7

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -25,6 +25,7 @@ platforms:
   - name: centos-7
   - name: debian-8
```
Comment (damacus) [inline .kitchen.yml]: shall we remove debian-8 support? 
I only really want to do, current-1

Comment (JohnRoesler) [reply .kitchen.yml]: Yeah I'll pull that out. Gotta fix the Deb 10 tests

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a <a href="https://github.com/sous-chefs/line-cookbook/blob/master/CHANGELOG.md">CHANGELOG</a> entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by JohnRoesler** [comment]
I'm a bit flummoxed. The `source-18-debian-10` test converges and passes locally with dokken. But fails in circleci.

**Comment by JohnRoesler** [comment]
@MarkGibbons yeah, you can converge with dokken using [this kitchen file](https://gist.github.com/JohnRoesler/b6998432a58884c691fbcd1a52fb241a) and `KITCHEN_YAML=.kitchen.dokken.yml kitchen converge source-18-debian-10` or run the test. You'll see if fails the service tests:

```
  Service haproxy
     ×  should be installed
     expected that `Service haproxy` is installed
     ×  should be enabled
     expected that `Service haproxy` is enabled
     ×  should be running
     expected that `Service haproxy` is running
```

**Comment by JohnRoesler** [comment]
closing this in favor of waiting for the official `stable` release of debian 10

**Pull request closed without merging**